app/assets/stylesheets/components/_modal-dialogue.scss

```diff
@@ -0,0 +1,146 @@
+$govuk-modal-margin: govuk-spacing(6);
+$govuk-modal-close-button-size: 40px;
+$govuk-modal-z-index: 1000;
+
+// Wrapper that the modal scrolls within
+.app-c-modal-dialogue {
+  display: block;
+  position: fixed;
+  z-index: $govuk-modal-z-index;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  overflow-x: hidden;
+  overflow-y: scroll;
+  outline: 0;
+}
+
+.app-c-modal-dialogue__box {
+  display: block;
+  position: fixed;
+  z-index: $govuk-modal-z-index - 10;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  margin: 0;
+  padding: 0;
+  overflow-y: auto;
+  border: 0;
+
+  @include govuk-media-query($from: tablet) {
+    position: relative;
+    top: inherit;
+    right: inherit;
+    bottom: inherit;
+    left: inherit;
+    width: auto;
+    max-width: $govuk-page-width * 2 / 3;
+    height: auto;
+    margin: $govuk-modal-margin auto;
+    border: $govuk-border-width-form-element solid $govuk-input-border-colour;
+  }
+
+  @include govuk-font($size: 19);
+  @include govuk-text-colour;
+  @include govuk-focusable;
+}
+
+.app-c-modal-dialogue__box--wide {
+  @include govuk-media-query($from: tablet) {
+    max-width: $govuk-page-width;
+  }
+}
+
+.app-c-modal-dialogue__overlay {
```

Comment:
Could this whole element be removed and replaced by putting `background: rgba(0,0,0,0.8)` on `app-c-modal-dialogue`?

Reply:
it could be, but we're also using this element to prevent clicking/focusing outside the modal with `pointer-events: none` - which feels more appropriate than overloading the component with additional scripts on events.